src/bridge/MatrixHandler.ts

```diff
@@ -547,14 +547,20 @@ export class MatrixHandler {
                     if (config && room.server.allowsNickChanges() &&
                         !config.getDesiredNick()
                     ) {
-                        try {
-                            const newNick = room.server.getNick(
-                                bridgedClient.userId, event.content.displayname
-                            );
-                            bridgedClient.changeNick(newNick, false);
-                        }
-                        catch (e) {
-                            req.log.warn(`Didn't change nick on the IRC side: ${e}`);
+                        const intent = this.ircBridge.getAppServiceBridge().getIntent();
+                        // Check that the /profile matches the displayname.
+                        const userProfile = await intent.getProfileInfo(event.state_key, "displayname");
```

Comment:
My understanding of the bridges is still poor, but looks like this code is only handled "on join"? So wondering shouldn't it be whenever the matrix side changes display name? And also if this is code that runs for everyone during a startup, will the profile details gets fetched for everyone or just changed names since previous restart?

Reply:
Ah, changing displaynames manifests as a resend of the m.room.member[content.membership="join"] event, which is handled by this function every time. onJoin is a slight misnomer

Reply:
>And also if this is code that runs for everyone during a startup, will the profile details gets fetched for everyone or just changed names since previous restart?

We shouldn't see a join for everyone during startup, so I don't think this is a huge concern.